enhancements/enhance-support-policy.md

```diff
@@ -0,0 +1,192 @@
+ * [Proposal for Pipelines &
+   GitOps](https://docs.google.com/spreadsheets/d/1Tn6BHNBJnEaTBRm22U0KrNCR_Zwu_qBn-IR-xFEa4Os/edit#gid=1430910332)
+
+## Motivation
+
+OpenShift Pipelines version are currently only shipped and supported
+for one specific version of OpenShift. This mean that, for example,
+OpenShift Pipelines 1.5 is only available and supported on 4.8,
+OpenShift Pipelines 1.6 on 4.9 and so on. Although this makes our
+"support life" easier, but it comes with some challenges:
+
+- this puts pressure on our customers to upgrade to the latest
+  OpenShift early if they want to benefit from the latest features that
+  are coming in OpenShift Pipelines.
+- this means our customers might lag behind for a long time and use
+  old version of upstream tekton components that have possibly bugs
+  already fixed.
+
+In addition, it has no real technological reason for not supporting
+multiple version. Tekton supports a relatively wide range of k8s
+version, and so can OpenShift Pipelines.
+
+Enabling customers to get the latest of feature from Tekton in their
+supported cluster is a win-win situation : customers can use the
+latest, and we can ship the latest (fixes, …) to our customers.
+
+## Challenges
+
+## OLM and contiguous supported ocp version
+
+As written in [SRVCOM-1654](https://issues.redhat.com/browse/SRVCOM-1654):
+
+> According to [Managing OpenShift
+> Versions](https://redhat-connect.gitbook.io/certified-operator-guide/ocp-deployment/operator-metadata/bundle-directory/managing-openshift-versions),
+> using a range will prevent the operator from being included in a
+> version past the highest version in the range. For example, the
+> Serverless 1.19.0 operator currently specifies "v4.6-v4.9". This
+> means that the operator will not be included in the OCP v4.10
+> index. If the operator had specified "v4.6" instead in the label,
+> then the 1.19.0 operator would be included in the OCP 4.10 index.
+
+If we switch to the “range” notation (as we started on nightlies), we
+will end up with forcing ourselves to release ahead of OpenShift each
+time if we want to not dig ourselves into a whole (aka, ending up with
+no version at a new OCP version released).
+
+The idea behind
+[SRVCOM-1654](https://issues.redhat.com/browse/SRVCOM-1654) is to set
+the ocp version at the lowest version that the team (serverless here)
+would like to support. Usually this would probably be the minimum
+supported version upstream. And each time that minimum supported
+version upstream is incremented, the ocp version supported would
+increase. On paper this could be seen as “too much” version to support
+as this could mean a version supporting 4.8 would be available in all
+future versions (if no further update). In practice, as we will
+release more versions, with the same “lower ocp version supported”, we
+would “hide” those versions and it would require a very explicit and
+“deep” knowledge to install a previous version of OpenShift Pipelines.
```

Comment:
So, today, with 1.6.x (and previous), we only specify one version of OpenShift. For OLM, this means : this version of this operator works on OpenShift 4.9 (for 1.6.x) and above. This means, *even* if we don't do anything, it will be available in the next release of OpenShift.

> How do we approach this? Either we do some preliminary testing (automated tests, etc) of the latest released OpenShift Pipelines on the latest released OCP so that it can be made available on any latest OpenShift versions?

Yes, that's exactly what we need to do. As soon as the next OpenShift versions has some builds (RCs or even before), we need to make sure our "supported" version of the Operator works on there. If it doesn't, we gotta look into fixing it without breaking the other versions.

> This `preview`/`bleeding`/`tech-preview` channel is _only_ visible on latest OCP versions for which we have not released any OpenShift Pipelines, yet. Once a fully supported release is made available, we can rename the`preview` channel to that respective release channel or (if possible via OLM) remove the `preview` channel and auto subscribe the user to the release channel.

I am not sure I get this, but, if I understood it right, it's honestly tricky to achieve, at least with the current "release" processes we have. For example, if we say 1.7 supports 4.9-4.11 (range notation), and we say it's available in `latest` and `pipelines-1.7` channels, then, it will be that way for the first 4.10 and 4.11 as well (because that's how the internal OLM/OCM RH toolchain will read our bundle). 

For example, having something in 1.7.0 in `latest` and `pipelines-1.7` channels in 4.10 **and** in `preview` for 4.11, means we need to do a first release for 4.10, and then do a *dot-release* of 1.7 but *just* publishing it in 4.11, on a different channel, and then re-do yet another *dot-release* for 4.9, 4.10 and 4.11 with the same set of channels. This force us to do even more releases, which I don't think we are ready for even. 
And what is the benefit of doing that then ? As soon as we make sure a given version (1.7 in our example) works on the upcoming OCP release, we are good, and I think it's relatively safe to assume we can do our job there.

The idea here, is, we declare, when we release a OSP version, which OCP version we support and will support, ahead of time. We have then to do our work and ensure it works on next version **and** fix it ahead of time if it doesn't.

Given the comment is on the "challenge" part, I want to make sure that it's understood that this particular sentences do not reflect the current *way* we will do things, as the "proposal" starts [here](https://github.com/openshift-pipelines/enhancements/pull/10/files#diff-207cf86593967bdd423ced80425bb4883519b15790f660669b7a0ca6be694265R111) (line 111).